socorro/signature/signature_utilities.py

```diff
@@ -2,31 +2,62 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from itertools import islice
+import logging
 import re
 
 import ujson
-from itertools import islice
 
-from configman import Namespace, RequiredConfig
-from configman.converters import class_converter
+from socorro import siglists
+from socorro.lib.treelib import tree_get
+
 
-from socorro.lib.transform_rules import Rule
+SIGNATURE_MAX_LENGTH = 255
+MAXIMUM_FRAMES_TO_CONSIDER = 40
+SIGNATURE_ESCAPE_SINGLE_QUOTE = True
 
-from socorro import siglists
 
+logger = logging.getLogger(__name__)
+
+
+class Rule(object):
+    """Base class for Signature generation rules"""
+    def predicate(self, raw_crash, processed_crash):
+        """Whether or not to run this rule
+
+        :arg dict raw_crash: the raw crash structure
+        :arg dict processed_crash: the processed crash structure
+
+        :returns: True or False
+
+        """
+        return True
 
-class SignatureTool(RequiredConfig):
+    def action(self, raw_crash, processed_crash, notes):
+        """Runs the rule against the data
+
+        This modifies ``processed_crash`` and ``notes`` in place.
+
+        :arg dict raw_crash: the raw crash structure
+        :arg dict processed_crash: the processed crash structure; modify the ``signature`` key
+        :arg list notes: list to add processing notes to
+
+        :returns: True
+
+        """
+        return True
```

Comment:
This is roughly based on the Socorro processor rules, but simplified in a few ways. I want to do another round of simplification and change `action` so that it doesn't mutate its arguments, but figured we can do that in a future phase if we wanted to. Doing it now would mean that even more things would be different and that'd likely make reviewing harder.